Sources/Rendering/OpenGL/ImageMapper/index.js

```diff
@@ -1310,6 +1318,24 @@ function vtkOpenGLImageMapper(publicAPI, model) {
         ptsArray[9] = spatialExt[1];
         ptsArray[10] = spatialExt[3];
         ptsArray[11] = sliceDepth;
+        if (sliceOffset === 0) {
+          // If the sliceOffset is 0, the scalars array is a view of the
+          // original scalars array. As a micro-optimization, we can get the
+          // range from the original scalars array and set the range to the
+          // scalars array.  This means there is no need to re-calculate the
+          // ranges for the scalars array.
```

Comment:
Isn't `imgScalars.getRange()` the range of the whole 3D volume, while `scalars` contains only a 2D slice ?

Reply:
I can't comment on this sorry.

Reply:
the problem is that it might be incorrect to use for a single slice the range computed over the whole 3D volume.

Reply:
I am far from an expert so feel free to suggest an improvement.  From my testing with cornerstone, my understanding is that `if (ijkMode === SlicingMode.K || ijkMode === SlicingMode.NONE)` AND `sliceOffset===0` then `scalars` is a direct copy of imgScalars. Is it possible to further narrow this check to guarantee that we can use the range? perhaps ` && sliceSize === basicScalars.length`?

Reply:
sliceSize is width*height, it is not width*height*depth. So it is not a is a direct copy of imgScalars, it is simply a direct subset of imgScalars but it IS NOT imgScalars. It is way smaller. So they may have a different "range".

However, for performance reasons, it can be argued that to use the "range of the volume" in place of the "range of a slice of the volume". For sure, we know that the range of a slice is at least within the range of its "volume". In a way, it does not make much sense that each slice gets a different range when we scroll along a volume. Therefore, we could decide that for `ImageMapper`, the range of the slice is never computed and the range of the volume is always used instead. That wouldn't be specific to the `ijkMode` value. It is just always the case.

Reply:
I would appreciate your input to advise the best option here.  

1. I can update `if (sliceOffset === 0) {` to `if (sliceOffset === 0 && scalars.length === basicScalars.length) {` ensuring the range is strictly valid
2. Leave the broader `if (sliceOffset === 0) {` as it is currently, justified as you have in your comment above.

Reply:
I would set the range with no condition. No matter what `ijkMode` nor `sliceOffset` worth.

Reply:
That is updated.